packages/shared/lib/utils/encryption.manager.ts

```diff
@@ -245,6 +248,10 @@ class EncryptionManager {
 
             if (!record.encryptedValue) {
                 decryptedDataRecords.push(dataRecord);
+                const nextCursor = record['_nango_metadata']['last_modified_at'] as unknown as string;
+                const id = dataRecord['id'];
+                const encodedCursorValue = Buffer.from(`${nextCursor}||${id}`).toString('base64');
+                record['_nango_metadata']['cursor'] = encodedCursorValue;
```

Comment:
I am not a huge fan of doing this in the encryption manager since it has nothing to do with encryption. IMHO this should live in `data.service`